packages/dds/tree/src/feature-libraries/editable-tree/utilities.ts

```diff
@@ -142,3 +159,99 @@ export function keyIsValidIndex(key: string | number, length: number): boolean {
     if (typeof key === "string" && String(index) !== key) return false;
     return Number.isInteger(index) && 0 <= index && index < length;
 }
+
+/**
+ * Attempts to wrap an arbitrary data into a cursor, trying to resolve data types according to the given schema.
+ *
+ * Note that the name of this function as well as its return type do not specify the tree format on purpose
+ * to encourage a development of support for pluggable formats/cursors in the EditableTree API.
+ */
+export function tryGetCursorFor(
+    schema: SchemaDataAndPolicy,
+    fieldSchema: FieldSchema,
+    data: unknown,
+): ITreeCursor {
+    const node = DetachedNode.create(schema, fieldSchema, data);
+    return singleMapTreeCursor(node);
+}
+
+/**
+ * This implementation of the MapTree is written to wrap an arbitrary data with respect to the tree schema.
+ *
+ * It is used as an intermediate storage for the user data,
+ * whenever a user utilizes simple assignments to change an EditableTree,
+ * as methods of the EditableTree accept only cursors as an input data.
+ */
+export class DetachedNode implements MapTree {
+    public readonly value?: TreeValue;
+    private readonly data?: object;
+
+    constructor(
+        public readonly schema: SchemaDataAndPolicy,
+        public readonly type: TreeSchemaIdentifier,
+        data: unknown,
+    ) {
+        if (isPrimitiveValue(data)) {
+            assertPrimitiveValueType(data, lookupTreeSchema(this.schema, this.type));
+            this.value = data;
+        } else {
+            assert(typeof data === "object" && data !== null, "Data should not be null.");
+            this.data = data;
+        }
+    }
+
+    /**
+     * Gets the fields of this node.
+     */
+    get fields(): Map<FieldKey, DetachedNode[]> {
```

Comment:
I think it is important to keep this reifying of the MapTree out of the user data lazy, as this follows the "shallow copy by reference" pattern of JS, so that if the user wraps the data into this class first, then changes the data outside the API, and only then uses this wrapper to change the EditableTree, the lazy loaded MapTree will reflect the changes done meanwhile (and let to re-use this even if the user data will be changed again afterwards). Also, it lets to keep this instance smaller in size by not immediately converting the user data into the MapTree.